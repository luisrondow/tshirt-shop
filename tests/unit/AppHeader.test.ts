import { describe, it, expect, beforeAll } from 'vitest'
import { shallowMount, RouterLinkStub, config } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'

import { AppHeader } from '#components'

describe('AppHeader component', () => {
  beforeAll(() => {
    config.global.renderStubDefaultSlot = true
  })

  it('should render correctly', () => {
    const wrapper = shallowMount(AppHeader, {
      stubs: {
        XLink: RouterLinkStub,
      },
      global: {
        plugins: [createTestingPinia()],
      },
    })

    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render logo, links and basket', () => {
    const wrapper = shallowMount(AppHeader, {
      stubs: {
        XLink: RouterLinkStub,
      },
      global: {
        plugins: [createTestingPinia()],
      },
    })

    const links = wrapper.findAll('x-link-stub')
    expect(links.length).toBe(3)
    expect(links[0].text()).toContain('Shop')
    expect(links[1].find('h1').text()).toBe('T-Shirt Shop')
    expect(links[2].text()).toContain('Basket')
  })
})
