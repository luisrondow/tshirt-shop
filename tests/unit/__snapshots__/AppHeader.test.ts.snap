// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppHeader component > should render correctly 1`] = `
<header
  class="flex w-full items-center justify-between px-2 py-4 md:px-6"
>
  <x-link-stub
    class="w-full uppercase"
    href=""
    to="/"
  >
    Shop
  </x-link-stub>
  <x-link-stub
    class="flex w-full justify-center !no-underline"
    href=""
    to="/"
  >
    <h1
      class="font-serif text-xl font-semibold md:text-4xl"
    >
      T-Shirt Shop
    </h1>
  </x-link-stub>
  <x-link-stub
    class="flex w-full cursor-pointer items-center justify-end gap-2"
    href=""
    to="/basket"
  >
    <span
      class="uppercase text-black hover:underline"
    >
      Basket
    </span>
    <div
      class="flex h-6 w-6 items-center justify-center rounded-full text-white bg-black"
      data-testid="basket-count"
    >
      0
    </div>
  </x-link-stub>
</header>
`;
